cmd/codegen/generator/go/templates/modules.go

```diff
@@ -642,6 +646,29 @@ func (ps *parseState) isMainModuleObject(name string) bool {
 	return strcase.ToCamel(ps.moduleName) == strcase.ToCamel(name)
 }
 
+// pkgDoc returns the package level documentation comment, if any.
+//
+// When there's multiple files with comments above the package declaration,
+// this returns the first one found, excluding from the generated file.
+func (ps *parseState) pkgDoc() string {
+	for _, syntax := range ps.pkg.Syntax {
+		for _, comment := range syntax.Comments {
+			// Skip comments that are below the package declaration.
+			if comment.Pos() > syntax.Package {
+				continue
+			}
+			// Skip comments that are in the generated file.
```

Comment:
Hm, what if we have multiple files all with their own comments?

This will only skip over the ones in `dagger.gen.go`. Not quite sure how we handle this for the other languages.

Reply:
> Another alternative here is to choose the file where the main struct is located

That sounds good to me. We can record the pos of the main struct in this loop: https://github.com/dagger/dagger/blob/dcee33ec84858610450ef30ddfcad60a9b9be053/cmd/codegen/generator/go/templates/modules.go#L106-L108

And then, we can pass that pos/file in to this func, and do a check like this: https://github.com/dagger/dagger/blob/dcee33ec84858610450ef30ddfcad60a9b9be053/cmd/codegen/generator/go/templates/modules.go#L655-L663

Reply:
Yeah, in Go it would probably be the file that has `main()`, but that's in the generated file 🙂 

I'll make the change, thanks!

Reply:
Fixed in latest commit.